### PR TITLE
Various Python 3.9 related import changes

### PR DESCRIPTION
**Pull request by hroncok**
See https://github.com/Theano/Theano/pull/6664#issuecomment-581845517

### CODE REVIEW

**Comment by nouiz** [comment]
This looks good to my. I won't test it manually. I'll wait a few days to let a chance to other people to comments too.

**Comment by hroncok** [comment]
There are some more things importe from collections directly, will update this.

File: theano/compat/__init__.py [hunk 1]
```diff
@@ -6,15 +6,16 @@
 from six import PY3, b, BytesIO, next
 from six.moves import configparser
 from six.moves import reload_module as reload
+from collections import OrderedDict
 try:
-    from collections.abc import (OrderedDict, MutableMapping as DictMixin,
-                                 Callable)
+    from collections.abc import Callable, Iterable, Mapping, ValuesView
+    from collections.abc import MutableMapping as DictMixin
```
Comment (twiecki) [inline theano/compat/__init__.py]: Why change the style syntax? I would just leave it as is with the parentheses.

Comment (hroncok) [reply theano/compat/__init__.py]: I can do that. The code was changed twice in the row (in the first commit, the import was shorter, in the second it turned out to be longer again, so I just did what I consider more readable).

Comment (hroncok) [reply theano/compat/__init__.py]: Rebased.

**Comment by nouiz** [comment]
It looks good to me and CI passed. So at least, it didn't cause regression in the Python version in our CI. @twiecki are you also good to merge that?

@hroncok did you ran Theano tests in Python 3.9? I would prefer to not have many PR that fix stuff in a new version in Python if possible.
Thanks

**Comment by hroncok** [comment]
> did you ran Theano tests in Python 3.9?

Yes.

[builder-live.log.gz](https://github.com/Theano/Theano/files/4159952/builder-live.log.gz)


You can add Python 3.9 to Travis CI.

**Comment by nouiz** [comment]
Thanks

**Pull request merged**